### DISCUSSION TITLE
Activity names issue

### DISCUSSION THREAD

**Discussion by Malicious1** [firdaus/temporal-python-sdk#6] [Ideas]
Hi,

I've found some issues regarding naming of activities (`activtyType.name` attribute). On the workflow side it is inferred from `ActivityInterfaceClassName::method_name` or you can give it a custom name by means of the `name` parameter of the `activity_method` decorator, thats all fine. 

Unfortunately on the activity  side (`register_activities_implementation`) you have very little room for setting the activityType and the two `::` are always there. So consider integration with a workflow code written in Java - you have to anticipate this issue and explicitly rename the activity interface so that it fits this convention. Have you considered allowing for passing the activityType.name directly to `register_activities_implementation`? Or maybe it is possible to somehow make use of the `name` param passed via the `activity_method` decorator (one would expect to inherit ActivitiesImplementation from ActivitiesInterface anyway right)?

**Comment by firdaus** [comment]
> Or maybe it is possible to somehow make use of the name param passed via the activity_method decorator (one would expect to inherit ActivitiesImplementation from ActivitiesInterface anyway right)?

This was how I thought I implemented it but I may have missed it. Thanks for alerting me.

**Comment by firdaus** [comment]
I filed an issue https://github.com/firdaus/temporal-python-sdk/issues/11

This is definitely something that needs to be resolved. Thank you for reporting it.
